cmd/posix.go

```diff
@@ -571,19 +600,61 @@ func (s *posix) ReadFile(volume string, path string, offset int64, buf []byte) (
 		return 0, err
 	}
 
-	// Verify if its not a regular file, since subsequent Seek is undefined.
+	// Verify if its not a regular file, since subsequent Seek is
+	// undefined.
 	if !st.Mode().IsRegular() {
 		return 0, errIsNotRegular
 	}
 
-	// Seek to requested offset.
-	_, err = file.Seek(offset, os.SEEK_SET)
-	if err != nil {
-		return 0, err
+	// If empty hash algo or an expected hash string is empty we
+	// do not perform hash verification.
+	needToHash := !(algo == HashAlgo("") || expectedHash == "")
```

Comment:
:+1:  good catch!

I've modified the behaviour now to skip hashing if and only if `expectedHash == ""`. If `algo == ""` then it defaults to using Blake2b hashing. This way there is no silent error.